repo2docker/buildpacks/base.py

```diff
@@ -481,13 +482,38 @@ def render(self):
             labels=self.get_labels(),
             build_script_directives=build_script_directives,
             assemble_script_directives=assemble_script_directives,
-            build_script_files=self.get_build_script_files(),
+            build_script_files={
+                self.generate_build_context_filename(k)[0]: v
+                for k, v in self.get_build_script_files().items()
+            },
             base_packages=sorted(self.get_base_packages()),
             post_build_scripts=self.get_post_build_scripts(),
             start_script=self.get_start_script(),
             appendix=self.appendix,
         )
 
+    @staticmethod
+    def generate_build_context_filename(src_path):
+        """
+        Generate a filename for a file injected into Docker build context.
+
+        In case the src_path is relative, it's assumed it's relative to directory of
+        this __file__. Returns the resulting filename and an absolute path to the source
+        file on host.
+        """
+        if not os.path.isabs(src_path):
+            src_parts = src_path.split('/')
+            src_path = os.path.join(os.path.dirname(__file__), *src_parts)
+        BLOCKSIZE = 65536
+        hash_md5 = hashlib.md5()
```

Comment:
I think we can get away with hashing the absolute path instead of having to hash the contents of the file. What do you think?

Reply:
Also let's use sha256, it should be fast enough - especially if we're just hashing the filename.

Reply:
I assumed that build scripts are small enough so that it won't matter, but I'm happy to oblige.

Reply:
Changed to `sha256` and computing slug using filepath only.